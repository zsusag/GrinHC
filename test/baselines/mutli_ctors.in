data Ctor =
 | First Float
 | Second Int Float
 | Third Int Float Int
 | Fourth Int Float Int Int

let z :: Ctor = Third 5 2.3 3
in case z of
  (First x) -> ( x )
  (Second x1 x2) -> ( x1 + x2 )
  (Third x1 x2 x3) -> ( x1 + x2 + x3 )
  (Fourth x1 x2 x3 x4) -> ( x1 + x2 + x3 + x4 )
  _ -> ( 3.2 )
end
