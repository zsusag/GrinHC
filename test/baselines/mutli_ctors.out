[("First",(TArr Float -> Ctor)),("Second",(TArr (TArr Int -> Float) -> Ctor)),("Third",(TArr (TArr (TArr Int -> Float) -> Int) -> Ctor)),("Fourth",(TArr (TArr (TArr (TArr Int -> Float) -> Int) -> Int) -> Ctor))]
10.3
